File: app/src/main/java/id/ilhamsuaib/androidprintbluetooth/BluetoothHandler.java
package id.ilhamsuaib.androidprintbluetooth;

import android.app.Activity;
import android.os.Handler;
import android.os.Message;
import android.util.Log;
import android.widget.Toast;

import com.zj.btsdk.BluetoothService;

/**
 * Created by ilham on 9/15/17.
 */

public class BluetoothHandler extends Handler{

    private static final String TAG = BluetoothHandler.class.getSimpleName();
    private Activity activity;
    private HandlerInterfce hi;

    public BluetoothHandler(Activity activity, HandlerInterfce hi) {
        this.activity = activity;
        this.hi = hi;
    }

    @Override
    public void handleMessage(Message msg) {
        switch (msg.what) {
            case BluetoothService.MESSAGE_STATE_CHANGE:
                switch (msg.arg1) {
                    case BluetoothService.STATE_CONNECTED:
                        hi.onDeviceConnected();
                        break;
                    case BluetoothService.STATE_CONNECTING:
                        hi.onDeviceConnecting();
                        break;
                    case BluetoothService.STATE_LISTEN:
                    case BluetoothService.STATE_NONE:
                        Log.i(TAG, "Bluetooth state listen or none");
                        break;
                }
                break;
            case BluetoothService.MESSAGE_CONNECTION_LOST:
                hi.onDeviceConnectionLost();
                break;
            case BluetoothService.MESSAGE_UNABLE_CONNECT:
                hi.onDeviceUnableToConnect();
                break;
        }
    }

    public interface HandlerInterfce {
        void onDeviceConnected();

        void onDeviceConnecting();

        void onDeviceConnectionLost();

        void onDeviceUnableToConnect();
    }
}

File: app/src/main/java/id/ilhamsuaib/androidprintbluetooth/BluetoothHandler.kt
package id.ilhamsuaib.androidprintbluetooth

import android.os.Handler
import android.os.Message
import android.util.Log

import com.zj.btsdk.BluetoothService

/**
 * Created by ilham on 9/15/17.
 */

class BluetoothHandler(private val hi: HandlerInterface) : Handler() {

    companion object {
        private const val TAG = "BluetoothHandler"
    }

    override fun handleMessage(msg: Message) {
        when (msg.what) {
            BluetoothService.MESSAGE_STATE_CHANGE -> when (msg.arg1) {
                BluetoothService.STATE_CONNECTED -> hi.onDeviceConnected()
                BluetoothService.STATE_CONNECTING -> hi.onDeviceConnecting()
                BluetoothService.STATE_LISTEN, BluetoothService.STATE_NONE -> Log.i(TAG, "Bluetooth state listen or none")
            }
            BluetoothService.MESSAGE_CONNECTION_LOST -> hi.onDeviceConnectionLost()
            BluetoothService.MESSAGE_UNABLE_CONNECT -> hi.onDeviceUnableToConnect()
        }
    }
}
